### PR TITLE
Private Events web application with user authentication using Rails Associations

### PR DESCRIPTION
**Pull request by emasdev**
## About Private Events
- Users can create events and sign to attendee an event. Events take place at a specific date and at a location.

## Goal
- The main goal is to demonstrate mastering in building Rails Associations

## Features

If you're not an authenticated user

- Create User
- Log in

If you are an authenticated user

- See the list of Events available divided by upcoming and previous events
- See users attendance list
- See the attendance list for a specific event
- Register to attend an event

Rspec Testing available

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2020_08_27_151052) do
+
+  create_table "event_attendings", force: :cascade do |t|
+    t.integer "attendee_id"
+    t.integer "attended_event_id"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
```
Comment (maelfosso) [inline db/schema.rb]: Please, add index for the referential fields.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      </div>
+
+      <div class="navbar-text">
+        <% if logged_in? %>
+          <span>You are Logged In, <%= current_user.name %></span> |
+          <%= link_to "Logout", controller: "sessions", action: "logout", :class => 'navbar-link' %>
+        <% else %>
+          <%= link_to "Login", '/login', method: :get, :class => 'navbar-link' %> |
+          <%= link_to "Sign Up", '/users/new', method: :get, :class => 'navbar-link' %>
+        <%end%>
```
Comment (maelfosso) [inline app/views/layouts/application.html.erb]: Remove logic from views please. implement them in view helpers.

File: app/controllers/events_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  before_action :logged_in?, except: %i[show index]
+
+  def index
+    @events = Event.all
+  end
+
+  def new
+    if current_user.nil?
+      redirect_to '/login'
+    else
```
Comment (maelfosso) [inline app/controllers/events_controller.rb]: No needed. Add the new method into the `before_action` statement

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,
Thank you for the job done so far. However, there are some missing requirements

1. Kindly add integration tests. It required. [Read it here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png)

2 Read the comments left below.

Kindly make the suggested changes  and submit a new code review request
Thanks

Mael FOSSO

**Comment by MarilenaRoque** [comment]
Hello @maelfosso. Thanks for the review, the correction, and the clarifications about the requirements. :+1: 
We worked in all the topics listed and all the changes were implemented.

- [x] Removed all the logic inside the view using helpers
- [x] Added index for all the foreign keys
- [x] Built integration tests
- [x] Removed the if for the redirection

# :rocket:

**Review by mosesogwo — APPROVED** [review]
**Hello!:wave: Good job with the changes :+1:**

You can now merge this PR.

_**Cheers:stars:**_

### Status: Approved :heavy_check_mark:

**Comment by MarilenaRoque** [comment]
# Thank you @mosesogwo! :rocket:

**Pull request merged**